src/Workspaces/Core/Portable/Workspace/Solution/AnalyzerConfigData.cs

```diff
@@ -23,9 +23,16 @@ internal readonly struct AnalyzerConfigData
     /// </summary>
     public readonly ImmutableDictionary<string, ReportDiagnostic> TreeOptions;
 
-    public AnalyzerConfigData(AnalyzerConfigOptionsResult result, StructuredAnalyzerConfigOptions fallbackOptions)
+    public AnalyzerConfigData(AnalyzerConfigOptionsResult result, StructuredAnalyzerConfigOptions fallbackOptions, bool legacyOverrideSuppressRazorSourceGenerator)
     {
-        _dictionaryConfigOptions = new DictionaryAnalyzerConfigOptions(result.AnalyzerOptions);
+        var entries = result.AnalyzerOptions;
+
+        if (legacyOverrideSuppressRazorSourceGenerator)
+        {
+            entries = entries.SetItem("build_property.SuppressRazorSourceGenerator", "false");
```

Comment:
I don't understand this bit. Why do we want to allow the generator to run if forceRuntimeCodeGen is on? Seems like that would break Razor IDE scenarios, because the IDE is going to provide C# via `IDynamicFileInfoProvider`, and the source generator will provide the same C#.

Reply:
We need to run the source generator for Hot Reload. Do we need to disable `IDynamicFileInfoProvider` if `forceRuntimeCodeGen` then?

Reply:
That would definitely break the IDE.

Maybe I'm missing context here,  it this feels like the PR we would want for cohosting, once https://github.com/dotnet/razor/pull/11412 is merged for example. That PR does indeed turn off `IDynamicFilInfo` in favour of the generator. Cohosting is a different feature flag though.

Force runtime code generation doesn't fundamentally change how Razor works in the IDE, it just makes the code generated look the same as what the source generator produces.

Reply:
I see, so we will need cohosting as well to make this work.